packages/connector-schemas/src/types.ts

```diff
@@ -0,0 +1,84 @@
+import type { Language } from '@logto/phrases';
+import { Nullable, Optional } from '@silverhand/essentials';
+
+import { ConnectorError, ConnectorErrorCodes } from './error';
+
+export enum ConnectorType {
+  Email = 'Email',
+  SMS = 'SMS',
+  Social = 'Social',
+}
+
+export enum ConnectorPlatform {
+  Native = 'Native',
+  Universal = 'Universal',
+  Web = 'Web',
+}
+
+type i18nPhrases = { [Language.English]: string } & {
+  [key in Exclude<Language, Language.English>]?: string;
+};
+
+export type ConnectorMetadata = {
+  id: string;
+  target: string;
+  type: ConnectorType;
+  platform: Nullable<ConnectorPlatform>;
+  name: i18nPhrases;
+  logo: string;
+  logoDark: Nullable<string>;
+  description: i18nPhrases;
+  readme: string;
+  configTemplate: string;
+};
+
+type MessageTypes = 'SignIn' | 'Register' | 'ForgotPassword' | 'Test';
```

Comment:
We'd better to use a same expression for those literals, either this way or enum.